common/lib/dependabot/dependency.rb

```diff
@@ -35,10 +35,12 @@ def self.register_name_normaliser(package_manager, name_builder)
       @name_normalisers[package_manager] = name_builder
     end
 
-    attr_reader :name, :version, :requirements, :package_manager,
+    attr_reader :name, :requirements, :package_manager,
                 :previous_version, :previous_requirements,
                 :subdependency_metadata
 
+    attr_accessor :version
```

Comment:
This is pretty out there eh?

It stems from the new `resolve_git_tags` helper - I wanted that to be called after the `DependencySet` had already de-duplicated entries, to minimize the amount of work it would have to do.
The YOLO solution was to make `Dependency#version` mutable, so the "resolve" step can re-assign it.

Reply:
I rolled this back to instead rely on [DependencySet#<<](https://github.com/dependabot/dependabot-core/blob/eb4f4779f9fa48553df060c5f580da14ce46b9fb/common/lib/dependabot/file_parsers/base/dependency_set.rb#L30-L35) to combine a `Dependency` with the resolved version with the actual `Dependency`.

It's a little gnarly, but it feels better than opening the Pandora's Box of mutability.